open-sphere-plugins/my-places/src/main/java/io/opensphere/myplaces/controllers/RoiMultiMenuProvider.java

```diff
@@ -0,0 +1,93 @@
+package io.opensphere.myplaces.controllers;
+
+import java.awt.Component;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import javax.swing.JMenuItem;
+
+import io.opensphere.core.Toolbox;
+import io.opensphere.core.control.action.ContextMenuProvider;
+import io.opensphere.core.geometry.Geometry;
+import io.opensphere.core.geometry.PolygonGeometry;
+import io.opensphere.core.util.collections.New;
+import io.opensphere.mantle.data.DataGroupInfo;
+import io.opensphere.mantle.data.DataTypeInfo;
+import io.opensphere.mantle.data.DataGroupInfo.MultiDataGroupContextKey;
+import io.opensphere.mantle.util.MantleToolboxUtils;
+import io.opensphere.myplaces.models.MyPlacesDataTypeInfo;
+import io.opensphere.myplaces.specific.regions.utils.RegionUtils;
+
+public class RoiMultiMenuProvider implements ContextMenuProvider<MultiDataGroupContextKey>
+{
+
+    /**
+     * The system toolbox.
+     */
+    private final Toolbox myToolbox;
+
+    /**
+     * Constructs a new ROI myplaces menu provider to be used for the context
+     * menu in the layers window.
+     *
+     * @param toolbox The system toolbox.
+     */
+    public RoiMultiMenuProvider(Toolbox toolbox)
+    {
+        myToolbox = toolbox;
+    }
+
+    @Override
+    public Collection<? extends Component> getMenuItems(String contextId, MultiDataGroupContextKey key)
+    {
+        List<JMenuItem> menuItems = Collections.emptyList();
+        Collection<Geometry> geom = New.list();
+        Collection<DataTypeInfo> dataTypes = getDataTypes(key);
+
+        if (!dataTypes.isEmpty())
+        {
+            for (DataTypeInfo dti : dataTypes)
+            {
+                if (dti instanceof MyPlacesDataTypeInfo)
+                {
+                    MyPlacesDataTypeInfo type = (MyPlacesDataTypeInfo)dti;
+                    PolygonGeometry theGeom = RegionUtils.createGeometry(type.getKmlPlacemark());
+
+                    geom.add(theGeom);
+                }
+            }
+            menuItems = MantleToolboxUtils.getMantleToolbox(myToolbox).getSelectionHandler().getMultiGeometryMenu(geom);
+        }
+
+        return menuItems;
+    }
+
+    @Override
+    public int getPriority()
+    {
+        // TODO Auto-generated method stub
+        return 0;
```

Comment:
Does the menu show up in a good spot?  Just wondering if priority should be set here or not.  Also remove the TODO comment.

Reply:
The 5 buttons show up in a good spot , since there is only 1 other button (Export)